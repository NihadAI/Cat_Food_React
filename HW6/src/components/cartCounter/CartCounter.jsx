import { useSelector } from 'react-redux';
import styles from './CartCounter.module.css';

function CartCounter() {
	const cartCounter = useSelector((state) => state.cart.count);

	return (
		<div className={styles.cart}>
			Cart <span className={styles.counter}>{cartCounter}</span>
		</div>
	);
}
export default CartCounter;
