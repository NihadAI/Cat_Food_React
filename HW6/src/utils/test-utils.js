import React from 'react';
import { render as rtlRender, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { MemoryRouter } from 'react-router-dom';

import currentsReducer from '../redux/slices/currentsSlice';
import cartReducer from '../redux/slices/cartSlice';
import favouritesReducer from '../redux/slices/favouritesSlice';
import modalReducer from '../redux/slices/modalSlice';
import itemsReducer from '../redux/slices/itemsSlice';

function render(ui, options = {}) {
	const initialState = {
		currents: { defaultScreenText: 'Test Screen Text' },
		modal: {
			isModalOpen: false,
			modalAction: {
				title: 'Test Title',
				content: 'Test Content',
				button: <button>Test button</button>,
			},
			isShadowDisabled: false,
		},
		cart: { items: [], count: 0, fullPrice: 0 },
		favourites: { items: [], count: 0 },
		items: { items: [], isLoading: true },
	};

	screen.debug();

	const testStore = configureStore({
		reducer: {
			currents: currentsReducer,
			cart: cartReducer,
			favourites: favouritesReducer,
			modal: modalReducer,
			items: itemsReducer,
		},
		preloadedState: initialState,
	});

	const Wrapper = ({ children }) => (
		<MemoryRouter initialEntries={['/test-route']}>
			<Provider store={testStore}>{children}</Provider>
		</MemoryRouter>
	);

	return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export { render, screen };
