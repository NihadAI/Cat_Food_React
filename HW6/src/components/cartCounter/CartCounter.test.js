import { render } from '../../utils/test-utils';
import CartCounter from '.';

describe('Smoke test', () => {
	it('should render', () => {
		const { asFragment } = render(<CartCounter />);

		expect(asFragment()).toMatchSnapshot();
	});
});
