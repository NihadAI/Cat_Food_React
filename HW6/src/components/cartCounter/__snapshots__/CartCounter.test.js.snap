// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke test should render 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    Cart 
    <span
      class="counter"
    >
      0
    </span>
  </div>
</DocumentFragment>
`;
